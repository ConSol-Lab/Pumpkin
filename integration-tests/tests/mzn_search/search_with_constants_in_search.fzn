% Taken from the model `areas.fzn` and the instance `3_3_3.dzn`
array [1..2] of int: X_INTRODUCED_40_ = [1,-1];
array [1..2] of int: X_INTRODUCED_99_ = [1,1];
array [1..5] of int: X_INTRODUCED_110_ = [1,1,1,1,1];
var 1..3: X_INTRODUCED_19_;
var {1,3}: X_INTRODUCED_21_;
var 2..3: X_INTRODUCED_22_;
var 2..3: X_INTRODUCED_23_;
var 2..3: X_INTRODUCED_26_;
var 1..4: X_INTRODUCED_29_;
var 1..4: X_INTRODUCED_31_;
var 1..4: X_INTRODUCED_32_;
var 1..4: X_INTRODUCED_33_;
var 1..4: X_INTRODUCED_35_;
var 1..4: X_INTRODUCED_36_;
var bool: X_INTRODUCED_37_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_41_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_42_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_43_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_45_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_46_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_47_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_49_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_50_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_51_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_53_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_54_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_55_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_56_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_57_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_58_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_59_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_60_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_61_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_63_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_64_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_66_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_67_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_68_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_70_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_71_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_73_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_74_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_75_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_77_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_78_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_79_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_80_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_81_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_82_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_83_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_84_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_85_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_86_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_87_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_88_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_90_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_92_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_93_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_94_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_95_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_96_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_97_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_100_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_101_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_102_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_103_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_104_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_105_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_106_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_107_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_108_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_111_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_112_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_113_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_114_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_115_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_116_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_117_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_118_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_119_ ::var_is_introduced :: is_defined_var;
var 0..1: X_INTRODUCED_120_ ::var_is_introduced :: is_defined_var;
var bool: X_INTRODUCED_89_ ::var_is_introduced  = true;
array [1..9] of var int: X_INTRODUCED_17_:: output_array([1..9]) = [1,X_INTRODUCED_19_,2,X_INTRODUCED_21_,X_INTRODUCED_22_,X_INTRODUCED_23_,3,3,X_INTRODUCED_26_];
constraint bool_clause([X_INTRODUCED_42_,X_INTRODUCED_46_,X_INTRODUCED_50_],[]);
constraint bool_clause([X_INTRODUCED_54_,X_INTRODUCED_57_,X_INTRODUCED_60_],[]);
constraint bool_clause([X_INTRODUCED_64_,X_INTRODUCED_67_,X_INTRODUCED_71_,X_INTRODUCED_74_],[]);
constraint bool_clause([X_INTRODUCED_78_,X_INTRODUCED_81_,X_INTRODUCED_83_],[]);
constraint bool_clause([X_INTRODUCED_85_,X_INTRODUCED_88_,X_INTRODUCED_90_],[]);
constraint bool_clause([X_INTRODUCED_93_,X_INTRODUCED_95_],[]);
constraint int_lin_eq(X_INTRODUCED_99_,[X_INTRODUCED_96_,X_INTRODUCED_97_],1);
constraint int_lin_eq(X_INTRODUCED_110_,[X_INTRODUCED_101_,X_INTRODUCED_103_,X_INTRODUCED_105_,X_INTRODUCED_106_,X_INTRODUCED_108_],2);
constraint int_lin_eq(X_INTRODUCED_110_,[X_INTRODUCED_112_,X_INTRODUCED_114_,X_INTRODUCED_116_,X_INTRODUCED_118_,X_INTRODUCED_120_],2);
constraint int_eq_reif(X_INTRODUCED_19_,X_INTRODUCED_22_,X_INTRODUCED_37_):: defines_var(X_INTRODUCED_37_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_29_,X_INTRODUCED_32_],1,X_INTRODUCED_41_):: defines_var(X_INTRODUCED_41_);
constraint array_bool_and([X_INTRODUCED_37_,X_INTRODUCED_41_],X_INTRODUCED_42_):: defines_var(X_INTRODUCED_42_);
constraint int_eq_reif(X_INTRODUCED_19_,2,X_INTRODUCED_43_):: defines_var(X_INTRODUCED_43_);
constraint int_lin_eq_reif([1],[X_INTRODUCED_29_],2,X_INTRODUCED_45_):: defines_var(X_INTRODUCED_45_);
constraint array_bool_and([X_INTRODUCED_43_,X_INTRODUCED_45_],X_INTRODUCED_46_):: defines_var(X_INTRODUCED_46_);
constraint int_eq_reif(X_INTRODUCED_19_,1,X_INTRODUCED_47_):: defines_var(X_INTRODUCED_47_);
constraint int_lin_eq_reif([1],[X_INTRODUCED_29_],2,X_INTRODUCED_49_):: defines_var(X_INTRODUCED_49_);
constraint array_bool_and([X_INTRODUCED_47_,X_INTRODUCED_49_],X_INTRODUCED_50_):: defines_var(X_INTRODUCED_50_);
constraint int_eq_reif(X_INTRODUCED_21_,3,X_INTRODUCED_51_):: defines_var(X_INTRODUCED_51_);
constraint int_lin_eq_reif([1],[X_INTRODUCED_31_],2,X_INTRODUCED_53_):: defines_var(X_INTRODUCED_53_);
constraint array_bool_and([X_INTRODUCED_51_,X_INTRODUCED_53_],X_INTRODUCED_54_):: defines_var(X_INTRODUCED_54_);
constraint int_eq_reif(X_INTRODUCED_21_,1,X_INTRODUCED_55_):: defines_var(X_INTRODUCED_55_);
constraint int_lin_eq_reif([1],[X_INTRODUCED_31_],2,X_INTRODUCED_56_):: defines_var(X_INTRODUCED_56_);
constraint array_bool_and([X_INTRODUCED_55_,X_INTRODUCED_56_],X_INTRODUCED_57_):: defines_var(X_INTRODUCED_57_);
constraint int_eq_reif(X_INTRODUCED_21_,X_INTRODUCED_22_,X_INTRODUCED_58_):: defines_var(X_INTRODUCED_58_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_31_,X_INTRODUCED_32_],1,X_INTRODUCED_59_):: defines_var(X_INTRODUCED_59_);
constraint array_bool_and([X_INTRODUCED_58_,X_INTRODUCED_59_],X_INTRODUCED_60_):: defines_var(X_INTRODUCED_60_);
constraint int_eq_reif(X_INTRODUCED_22_,3,X_INTRODUCED_61_):: defines_var(X_INTRODUCED_61_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_32_,X_INTRODUCED_35_],1,X_INTRODUCED_63_):: defines_var(X_INTRODUCED_63_);
constraint array_bool_and([X_INTRODUCED_61_,X_INTRODUCED_63_],X_INTRODUCED_64_):: defines_var(X_INTRODUCED_64_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_32_,X_INTRODUCED_29_],1,X_INTRODUCED_66_):: defines_var(X_INTRODUCED_66_);
constraint array_bool_and([X_INTRODUCED_37_,X_INTRODUCED_66_],X_INTRODUCED_67_):: defines_var(X_INTRODUCED_67_);
constraint int_eq_reif(X_INTRODUCED_22_,X_INTRODUCED_23_,X_INTRODUCED_68_):: defines_var(X_INTRODUCED_68_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_32_,X_INTRODUCED_33_],1,X_INTRODUCED_70_):: defines_var(X_INTRODUCED_70_);
constraint array_bool_and([X_INTRODUCED_68_,X_INTRODUCED_70_],X_INTRODUCED_71_):: defines_var(X_INTRODUCED_71_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_32_,X_INTRODUCED_31_],1,X_INTRODUCED_73_):: defines_var(X_INTRODUCED_73_);
constraint array_bool_and([X_INTRODUCED_58_,X_INTRODUCED_73_],X_INTRODUCED_74_):: defines_var(X_INTRODUCED_74_);
constraint int_eq_reif(X_INTRODUCED_23_,X_INTRODUCED_26_,X_INTRODUCED_75_):: defines_var(X_INTRODUCED_75_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_33_,X_INTRODUCED_36_],1,X_INTRODUCED_77_):: defines_var(X_INTRODUCED_77_);
constraint array_bool_and([X_INTRODUCED_75_,X_INTRODUCED_77_],X_INTRODUCED_78_):: defines_var(X_INTRODUCED_78_);
constraint int_eq_reif(X_INTRODUCED_23_,2,X_INTRODUCED_79_):: defines_var(X_INTRODUCED_79_);
constraint int_lin_eq_reif([1],[X_INTRODUCED_33_],2,X_INTRODUCED_80_):: defines_var(X_INTRODUCED_80_);
constraint array_bool_and([X_INTRODUCED_79_,X_INTRODUCED_80_],X_INTRODUCED_81_):: defines_var(X_INTRODUCED_81_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_33_,X_INTRODUCED_32_],1,X_INTRODUCED_82_):: defines_var(X_INTRODUCED_82_);
constraint array_bool_and([X_INTRODUCED_68_,X_INTRODUCED_82_],X_INTRODUCED_83_):: defines_var(X_INTRODUCED_83_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_35_,X_INTRODUCED_32_],1,X_INTRODUCED_84_):: defines_var(X_INTRODUCED_84_);
constraint array_bool_and([X_INTRODUCED_61_,X_INTRODUCED_84_],X_INTRODUCED_85_):: defines_var(X_INTRODUCED_85_);
constraint int_eq_reif(3,X_INTRODUCED_26_,X_INTRODUCED_86_):: defines_var(X_INTRODUCED_86_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_35_,X_INTRODUCED_36_],1,X_INTRODUCED_87_):: defines_var(X_INTRODUCED_87_);
constraint array_bool_and([X_INTRODUCED_86_,X_INTRODUCED_87_],X_INTRODUCED_88_):: defines_var(X_INTRODUCED_88_);
constraint int_lin_eq_reif([1],[X_INTRODUCED_35_],2,X_INTRODUCED_90_):: defines_var(X_INTRODUCED_90_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_36_,X_INTRODUCED_33_],1,X_INTRODUCED_92_):: defines_var(X_INTRODUCED_92_);
constraint array_bool_and([X_INTRODUCED_75_,X_INTRODUCED_92_],X_INTRODUCED_93_):: defines_var(X_INTRODUCED_93_);
constraint int_lin_eq_reif(X_INTRODUCED_40_,[X_INTRODUCED_36_,X_INTRODUCED_35_],1,X_INTRODUCED_94_):: defines_var(X_INTRODUCED_94_);
constraint array_bool_and([X_INTRODUCED_86_,X_INTRODUCED_94_],X_INTRODUCED_95_):: defines_var(X_INTRODUCED_95_);
constraint bool2int(X_INTRODUCED_47_,X_INTRODUCED_96_):: defines_var(X_INTRODUCED_96_);
constraint bool2int(X_INTRODUCED_55_,X_INTRODUCED_97_):: defines_var(X_INTRODUCED_97_);
constraint int_eq_reif(X_INTRODUCED_19_,2,X_INTRODUCED_100_):: defines_var(X_INTRODUCED_100_);
constraint bool2int(X_INTRODUCED_100_,X_INTRODUCED_101_):: defines_var(X_INTRODUCED_101_);
constraint int_eq_reif(X_INTRODUCED_21_,2,X_INTRODUCED_102_):: defines_var(X_INTRODUCED_102_);
constraint bool2int(X_INTRODUCED_102_,X_INTRODUCED_103_):: defines_var(X_INTRODUCED_103_);
constraint int_eq_reif(X_INTRODUCED_22_,2,X_INTRODUCED_104_):: defines_var(X_INTRODUCED_104_);
constraint bool2int(X_INTRODUCED_104_,X_INTRODUCED_105_):: defines_var(X_INTRODUCED_105_);
constraint bool2int(X_INTRODUCED_79_,X_INTRODUCED_106_):: defines_var(X_INTRODUCED_106_);
constraint int_eq_reif(X_INTRODUCED_26_,2,X_INTRODUCED_107_):: defines_var(X_INTRODUCED_107_);
constraint bool2int(X_INTRODUCED_107_,X_INTRODUCED_108_):: defines_var(X_INTRODUCED_108_);
constraint int_eq_reif(X_INTRODUCED_19_,3,X_INTRODUCED_111_):: defines_var(X_INTRODUCED_111_);
constraint bool2int(X_INTRODUCED_111_,X_INTRODUCED_112_):: defines_var(X_INTRODUCED_112_);
constraint int_eq_reif(X_INTRODUCED_21_,3,X_INTRODUCED_113_):: defines_var(X_INTRODUCED_113_);
constraint bool2int(X_INTRODUCED_113_,X_INTRODUCED_114_):: defines_var(X_INTRODUCED_114_);
constraint int_eq_reif(X_INTRODUCED_22_,3,X_INTRODUCED_115_):: defines_var(X_INTRODUCED_115_);
constraint bool2int(X_INTRODUCED_115_,X_INTRODUCED_116_):: defines_var(X_INTRODUCED_116_);
constraint int_eq_reif(X_INTRODUCED_23_,3,X_INTRODUCED_117_):: defines_var(X_INTRODUCED_117_);
constraint bool2int(X_INTRODUCED_117_,X_INTRODUCED_118_):: defines_var(X_INTRODUCED_118_);
constraint int_eq_reif(X_INTRODUCED_26_,3,X_INTRODUCED_119_):: defines_var(X_INTRODUCED_119_);
constraint bool2int(X_INTRODUCED_119_,X_INTRODUCED_120_):: defines_var(X_INTRODUCED_120_);
solve :: int_search([1,X_INTRODUCED_19_,2,X_INTRODUCED_21_,X_INTRODUCED_22_,X_INTRODUCED_23_,3,3,X_INTRODUCED_26_],first_fail,indomain_split,complete) satisfy;
