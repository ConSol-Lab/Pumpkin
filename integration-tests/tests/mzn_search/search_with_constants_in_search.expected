X_INTRODUCED_17_ = array1d(1..9,[1, 1, 2, 3, 3, 2, 3, 3, 2]);
----------
X_INTRODUCED_17_ = array1d(1..9,[1, 1, 2, 3, 2, 2, 3, 3, 3]);
----------
X_INTRODUCED_17_ = array1d(1..9,[1, 2, 2, 1, 3, 2, 3, 3, 3]);
----------
X_INTRODUCED_17_ = array1d(1..9,[1, 2, 2, 1, 2, 3, 3, 3, 3]);
----------
X_INTRODUCED_17_ = array1d(1..9,[1, 3, 2, 1, 3, 2, 3, 3, 2]);
----------
==========